internal/cli/atlas/accesslists/create.go

```diff
@@ -80,26 +83,81 @@ func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
 	return []*atlas.ProjectIPAccessList{entry}
 }
 
+func iPAddress() (string, error) {
+	if publicIP := store.IPAddress(); publicIP != "" {
+		return publicIP, nil
+	}
+
+	err := fmt.Errorf(currentIPAddressNotFound)
+	return "", err
+}
+
+func needsArg(opts *CreateOpts) bool {
+	// Unless currentIP flag is enabled and type is ip address, args are required.
+	return !(opts.entryType == ipAddress && opts.currentIP)
+}
+
+func (opts *CreateOpts) validateCurrentIPFlag(cmd *cobra.Command, args []string) func() error {
+	if !needsArg(opts) && len(args) > 0 {
+		return func() error {
+			return fmt.Errorf(
+				"please either provide <entry> or use %s to use current IP Address.\n\nUsage: %s",
+				flag.CurrentIP,
+				cmd.UseLine(),
+			)
+		}
+	}
+
+	if needsArg(opts) && len(args) == 0 {
+		return func() error {
+			return fmt.Errorf(
+				"%q with entry type %s requires at least %d argument received %d\n\nUsage:  %s",
+				cmd.CommandPath(),
+				opts.entryType,
+				1,
+				len(args),
+				cmd.UseLine(),
+			)
+		}
+	}
+	return func() error { return nil }
+}
+
 // mongocli atlas accessList(s) create <entry> --type cidrBlock|ipAddress|awsSecurityGroup [--comment comment] [--projectId projectId].
 func CreateBuilder() *cobra.Command {
 	opts := &CreateOpts{}
 	cmd := &cobra.Command{
 		Use:   "create <entry>",
```

Comment:
[nit] for POSIX `<>` defines a required arg, since entry i snow optional let's switch it to `[]`

Reply:
nice catch. thanks!